### DISCUSSION TITLE
Operation of syncUrl

### DISCUSSION THREAD

**Discussion by ShaunBrassell** [HaylLtd/cordova-plugin-background-geolocation#116] [General]
Morning Harel

Trying to figure out the best way to handle bad connectivity and failed uploads. Is this understanding correct:

If an upload to server fails, it is will be uploaded via the syncUrl url when the number of failed messages >= the syncThreshold?
It will also upload if I run the forceSync method at any time?

The functionality I am trying to achieve is that when connectivity is good again, any failed to upload messages are then uploaded as soon as can be. If I set the syncThreshold to say 5, if I have 4 failures and then all good again, will these 4 not upload as there are no more failures to take the threshold to 5 in order to trigger the syncUrl?

Thanks man
Shaun

**Comment by HarelM** [comment]
Unfortunately, I'm not using the http functionality of this plugin so I don't have a lot of info and knowledge to share here.
You'll need to dig into this and understand how this mechanism works...
